src/VisualStudio/Core/Def/Implementation/SymbolSearch/SymbolSearchResult.cs

```diff
@@ -0,0 +1,232 @@
+        {
+            var (projectGuid, documentGuid, projectName, sourceText) = await FindUsagesUtilities.GetGuidAndProjectNameAndSourceTextAsync(documentSpan.Document, cancellationToken)
+                .ConfigureAwait(false);
+
+            var (excerptResult, lineText) = await FindUsagesUtilities.ExcerptAsync(sourceText, documentSpan, cancellationToken).ConfigureAwait(false);
+            var plainText = excerptResult.Content.ToString();
+            var result = new RoslynSymbolSearchResult(context, plainText);
+
+            result.ProjectGuid = projectGuid.ToString();
+            result.DocumentGuid = documentGuid.ToString();
+            result.ProjectName = projectName;
+
+            if (referenceItem != null)
+            {
+                var classifiedSpansAndHighlightSpan = await ClassifiedSpansAndHighlightSpanFactory.ClassifyAsync(referenceItem.SourceSpan, cancellationToken).ConfigureAwait(false);
+                var classifiedSpans = classifiedSpansAndHighlightSpan.ClassifiedSpans;
+                var docText = await referenceItem.SourceSpan.Document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                result.ContextText = new ClassifiedTextElement(classifiedSpans.Select(cspan => new ClassifiedTextRun(cspan.ClassificationType, docText.ToString(cspan.TextSpan))));
+                result.ContextHighlightSpan = AsSpan(classifiedSpansAndHighlightSpan.HighlightSpan);
+
+                if (referenceItem.AdditionalProperties.TryGetValue(AbstractReferenceFinder.ContainingTypeInfoPropertyName, out var containingTypeInfo))
+                {
+                    result.ContainingTypeName = containingTypeInfo;
+                }
+
+                if (referenceItem.AdditionalProperties.TryGetValue(AbstractReferenceFinder.ContainingMemberInfoPropertyName, out var containingMemberInfo))
+                {
+                    result.ContainingMemberName = containingMemberInfo;
+                }
+
+                // Get read\written information
+                if (referenceItem.IsWrittenTo)
+                {
+                    result.IsWrittenTo = true;
+                }
+                else
+                {
+                    result.IsReadFrom = true;
+                }
+
+                // Override read\writtern information from SymbolUsageInfo
+                if (referenceItem.SymbolUsageInfo.IsWrittenTo() && referenceItem.SymbolUsageInfo.IsReadFrom())
+                {
+                    result.IsReadFrom = true;
+                    result.IsWrittenTo = true;
+                }
+                else if (referenceItem.SymbolUsageInfo.IsWrittenTo())
+                {
+                    result.IsReadFrom = false;
+                    result.IsWrittenTo = true;
+                }
+                else if (referenceItem.SymbolUsageInfo.IsReadFrom())
+                {
+                    result.IsReadFrom = true;
+                    result.IsWrittenTo = false;
+                }
```

Comment:
this entire chain feels like it can be replaced with `result.IsReadFrom = ...isReadFrom(); result.IsWrittenTo = ...IsWrittenTo();`